metadata-ingestion/tests/unit/api/source_helpers/test_source_helpers.py

```diff
@@ -656,3 +661,79 @@ def test_auto_empty_dataset_usage_statistics_invalid_timestamp(
             changeType=models.ChangeTypeClass.CREATE,
         ).as_workunit(),
     ]
+
+
+@freeze_time("2023-01-02 00:00:00")
+def test_auto_patch_last_modified():
+    mcps = [
+        MetadataChangeProposalWrapper(
+            entityUrn="urn:li:container:008e111aa1d250dd52e0fd5d4b307b1a",
+            aspect=models.StatusClass(removed=False),
+        )
+    ]
+
+    initial_wu = list(auto_workunit(mcps))
+
+    expected = initial_wu
+
+    assert (
+        list(auto_patch_last_modified(initial_wu)) == expected
+    )  # There should be no change
+
+    mcps.append(
+        MetadataChangeProposalWrapper(
+            entityUrn="urn:li:dataset:a.b.c",
+            aspect=models.OperationClass(
+                timestampMillis=10,
+                lastUpdatedTimestamp=12,
+                operationType=OperationTypeClass.CREATE,
+            ),
+        )
+    )
+
+    mcps.append(
+        MetadataChangeProposalWrapper(
+            entityUrn="urn:li:dataset:a.b.c",
+            aspect=models.OperationClass(
+                timestampMillis=11,
+                lastUpdatedTimestamp=20,
+                operationType=OperationTypeClass.CREATE,
+            ),
+        )
+    )
+
+    expected = list(auto_workunit(mcps))
+
+    dataset_patch_builder = DatasetPatchBuilder(
+        urn="urn:li:dataset:a.b.c"
+    ).set_last_modified(TimeStampClass(time=20))
```

Comment:
need a test where the inputs into the helper has a MCP generated by DatasetPatchBuilder